astropy/config/configuration.py

```diff
@@ -521,7 +521,7 @@ def get_config(packageormod=None, reload=False):
     cobj = _cfgobjs.get(rootname, None)
 
     if cobj is None or reload:
-        if _ASTROPY_SETUP_:
+        if _ASTROPY_SETUP_ or "ASTROPY_SUPPRESS_CONFIG" in environ:
```

Comment:
"suppress config" here is confusing. What it appears to be really doing is to "override default config" with some non-standard one. I recommend a rename of this env var.